pkg/tenant/projects_test.go

```diff
@@ -1 +1,163 @@
+		TenantID: ulid.MustParse("01GMTWFK4XZY597Y128KXQ4WHP"),
+		ID:       ulid.MustParse("01GKKYAWC4PA72YC53RVXAEC67"),
+		Name:     "project-example",
+	}
+
+	defer cancel()
+
+	// Connect to mock trtl database.
+	trtl := db.GetMock()
+	defer trtl.Reset()
+
+	// Marshal the project data with msgpack.
+	data, err := project.MarshalValue()
+	require.NoError(err, "could not marshal the project")
+
+	// Unmarshal the project data with msgpack.
+	other := &db.Project{}
+	err = other.UnmarshalValue(data)
+	require.NoError(err, "could not unmarshal the project")
+
+	// Call the OnGet method and return the test data.
+	trtl.OnGet = func(ctx context.Context, gr *pb.GetRequest) (*pb.GetReply, error) {
+		return &pb.GetReply{
+			Value: data,
+		}, nil
+	}
+
+	// Should return an error if the project does not exist.
+	_, err = suite.client.ProjectDetail(ctx, "invalid")
+	suite.requireError(err, http.StatusBadRequest, "could not parse project ulid", "expected error when project does not exist")
+
+	// Call the OnPut method and return a PutReply.
+	trtl.OnPut = func(ctx context.Context, pr *pb.PutRequest) (*pb.PutReply, error) {
+		return &pb.PutReply{}, nil
+	}
+
+	// Should return an error if the project name does not exist.
+	_, err = suite.client.ProjectUpdate(ctx, &api.Project{ID: "01GKKYAWC4PA72YC53RVXAEC67"})
+	suite.requireError(err, http.StatusBadRequest, "project name is required", "expected error when project name does not exist")
+
+	req := &api.Project{
+		ID:   "01GKKYAWC4PA72YC53RVXAEC67",
+		Name: "project-example",
+	}
+
+	rep, err := suite.client.ProjectUpdate(ctx, req)
+	require.NoError(err, "could not update project")
+	require.NotEqual(req.ID, "01GMTWFK4XZY597Y128KXQ4WHP", "project id should not match")
+	require.Equal(rep.Name, req.Name, "expected project name to match")
+
+	// Should return an error if the project ID is parsed but not found.
+	trtl.OnGet = func(ctx context.Context, gr *pb.GetRequest) (*pb.GetReply, error) {
+		return nil, errors.New("key not found")
+	}
+
+	_, err = suite.client.ProjectDetail(ctx, "01GKKYAWC4PA72YC53RVXAEC67")
+	suite.requireError(err, http.StatusNotFound, "could not retrieve project", "expected error when project ID is not found")
+}
+
+func (suite *tenantTestSuite) TestProjectDelete() {
+	require := suite.Require()
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	projectID := "01GKKYAWC4PA72YC53RVXAEC67"
+
+	defer cancel()
+
+	// Connect to mock trtl database.
+	trtl := db.GetMock()
+	defer trtl.Reset()
+
+	// Call the OnDelete method and return a DeleteReply.
+	trtl.OnDelete = func(ctx context.Context, dr *pb.DeleteRequest) (*pb.DeleteReply, error) {
+		return &pb.DeleteReply{}, nil
+	}
+
+	// Should return an error if the project does not exist.
+	err := suite.client.ProjectDelete(ctx, "invalid")
+	suite.requireError(err, http.StatusBadRequest, "could not parse project ulid", "expected error when project does not exist")
+
+	err = suite.client.MemberDelete(ctx, projectID)
```

Comment:
I believe this should be `ProjectDelete`

Reply:
Great catch!